cmd/harmony/main.go

```diff
@@ -510,6 +510,15 @@ func setupNodeAndRun(hc harmonyconfig.HarmonyConfig) {
 
 	go listenOSSigAndShutDown(currentNode)
 
+	if hc.Network.NetworkType == nodeconfig.Partner { // devnet
+		go func() {
+			for {
+				<-time.After(10 * time.Second)
+				go currentNode.BroadcastCrossLinkFromShardsToBeacon()
```

Comment:
why do we need this here ?

Reply:
`BroadcastCrossLinkFromShardsToBeacon` happens only on a new block insertion. There is infinity view change, no new blocks.

Reply:
It's not always the case that we are in an infinite view change. Under normal conditions (i.e., no view change), could there be any race conditions here, or what happens if we broadcasting a crosslink twice?

Question is why would a shard need to broadcast a crosslink when it is not producing block?

Reply:
> what happens if we broadcasting a crosslink twice

crosslink broadcasted by leader and 1% of validators. It's duplicated by design. 

> why would a shard need to broadcast a crosslink when it is not producing block

Because shard 0 broadcasts struct `CrosslinkHeartbeat` with field `LatestContinuousBlockNum`. It knows about last blocks received by crosslink, but crosslink should be sent first. Actually, that is the second issue, where we can't verify `CrosslinkHeartbeat` message, because no key in committee, which is solved [here](https://github.com/harmony-one/harmony/pull/4669)

so, there is circular dependency, no heartbeat because no crosslinks, but no crosslinks, because no heartbeat.